LanguageExt.Core/DataTypes/UnitsOfMeasure/Mass.cs

```diff
@@ -0,0 +1,174 @@
+
+    #region Basic maths functions
+
+    public Mass Add(Mass rhs) =>
+        new Mass(Kilograms + rhs.Kilograms);
+
+    public Mass Subtract(Mass rhs) =>
+        new Mass(Kilograms - rhs.Kilograms);
+
+    public Mass Multiply(double rhs) =>
+        new Mass(Kilograms * rhs);
+
+    public Mass Divide(double rhs) =>
+        new Mass(Kilograms / rhs);
+
+    #endregion
+
+    #region Operators
+
+    public static Mass operator *(Mass lhs, double rhs) =>
+        lhs.Multiply(rhs);
+
+    public static Mass operator *(double lhs, Mass rhs) =>
+        rhs.Multiply(lhs);
+
+    public static Mass operator +(Mass lhs, Mass rhs) =>
+        lhs.Add(rhs);
+
+    public static Mass operator -(Mass lhs, Mass rhs) =>
+        lhs.Subtract(rhs);
+
+    public static Mass operator /(Mass lhs, double rhs) =>
+        lhs.Divide(rhs);
+
+    public static double operator /(Mass lhs, Mass rhs) =>
+        lhs.Kilograms / rhs.Kilograms;
+
+    public static bool operator ==(Mass lhs, Mass rhs) =>
+        lhs.Equals(rhs);
+
+    public static bool operator !=(Mass lhs, Mass rhs) =>
+        !lhs.Equals(rhs);
+
+    public static bool operator >(Mass lhs, Mass rhs) =>
+        lhs.Kilograms > rhs.Kilograms;
+
+    public static bool operator <(Mass lhs, Mass rhs) =>
+        lhs.Kilograms < rhs.Kilograms;
+
+    public static bool operator >=(Mass lhs, Mass rhs) =>
+        lhs.Kilograms >= rhs.Kilograms;
+
+    public static bool operator <=(Mass lhs, Mass rhs) =>
+        lhs.Kilograms <= rhs.Kilograms;
+
+    public Mass Round() =>
+        new Mass(Math.Round(Kilograms));
+
+    public Mass Sqrt() =>
+        new Mass(Math.Sqrt(Kilograms));
+
+    public Mass Abs() =>
+        new Mass(Math.Abs(Kilograms));
+
+    public Mass Min(Mass rhs) =>
+        new Mass(Math.Min(Kilograms, rhs.Kilograms));
+
+    public Mass Max(Mass rhs) =>
+        new Mass(Math.Max(Kilograms, rhs.Kilograms));
+
+    #endregion
+
+    #region Conversion units
+
+    public double Grams => Value / 1000;
+    public double Kilograms => Value;
+    public double Tonnes => Value * 1000;
+
+    public double Ounces => Pounds * 16;
+    public double Pounds => Value * 2.2046226;
+    public double Stones => Pounds / 14;
+    public double TonsUK => Value / 0.00098421;
+    public double TonsUS => Value / 0.0011023;
+
+    #endregion
+  }
+
+  public static class UnitsMassExtensions
+  {
+    public static Mass Grams(this int self)
+        => new Mass(self / 1000.0);
+    public static Mass Grams(this double self)
+        => new Mass(self / 1000);
+    public static Mass Grams(this float self)
+        => new Mass(self / 1000);
+    public static Mass Kilograms(this int self)
+        => new Mass(self);
+    public static Mass Kilograms(this double self)
+        => new Mass(self);
+    public static Mass Kilograms(this float self)
+        => new Mass(self);
+    public static Mass Tonnes(this int self)
+        => new Mass(self * 1000);
```

Comment:
Safer when dealing with `int` values to always make the constant `1000.0` to not lose values on integer overflow that a `double` could have _inexactly_ represented.